src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,38 @@
 package mate.academy;
 
+import java.util.HashMap;
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
+    private static final int WAIT_MILLISECONDS = 1000;
     private final Executor executor;
+    private final Map<String, UserData> cache = new HashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        if (cache.containsKey(userId)) {
+            return CompletableFuture.completedFuture(cache.get(userId));
+        }
+
+        return CompletableFuture.supplyAsync(() -> {
```

Comment:
Using 'supplyAsync' without specifying an error handling mechanism can lead to unhandled exceptions if the computation throws an error. It's recommended to handle exceptions using 'handle', 'exceptionally', or 'whenComplete' methods.